orbit/cmd/desktop/desktop.go

```diff
@@ -438,17 +440,36 @@ func main() {
 	// FIXME: it doesn't look like this is actually triggering, at least when desktop gets
 	// killed (https://github.com/fleetdm/fleet/issues/21256)
 	onExit := func() {
-		log.Info().Msg("exit")
+		log.Info().Msg("exiting")
 		if mdmMigrator != nil {
+			log.Debug().Err(err).Msg("exiting mdmMigrator")
 			mdmMigrator.Exit()
 		}
 		if swiftDialogCh != nil {
+			log.Debug().Err(err).Msg("exiting swiftDialogCh")
 			close(swiftDialogCh)
 		}
+		log.Debug().Msg("stopping ticker")
 		summaryTicker.Stop()
+		log.Debug().Msg("canceling offline watcher ctx")
 		cancelOfflineWatcherCtx()
 	}
 
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(
+		sigChan,
+		syscall.SIGINT,
+		syscall.SIGTERM,
+		syscall.SIGQUIT,
+	)
+
+	// Catch signals and exit gracefully
+	go func() {
+		s := <-sigChan
+		log.Info().Stringer("signal", s).Msg("Caught signal, exiting")
+		systray.Quit()
+	}()
```

Comment:
Double checking: Do we want to do this on the three OSs? (syscall.SIGTERM is no-op for Windows, right?)

Reply:
I believe it's just a mask, I'll run a quick test to check 🙂

Reply:
Looks like windows has no problem compiling, it just ignores the ones that don't do anything

Reply:
Another double check, is this change necessary for Windows?, see `setupRunners`.